src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+    private int capacity;
+
     @Override
-    public void add(T value) {
+    protected Object clone() throws CloneNotSupportedException {
+        return super.clone();
+    }
 
+    public void growCapacity() {
```

Comment:
```suggestion
    private void resize() {
```
Also, private method should be placed after all public methods
Think how to modify this logic according to comment above?